src/o365/graph/commands/siteclassification/siteclassification-set.ts

```diff
@@ -0,0 +1,267 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import * as request from 'request-promise-native';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate, CommandError
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../GraphCommand';
+import { DirectorySetting, UpdateDirectorySetting } from './DirectorySetting';
+import { DirectorySettingValue } from './DirectorySettingValue';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  classifications?: string;
+  defaultClassification?: string;
+  usageGuidelinesUrl?: string;
+  guestUsageGuidelinesUrl?: string;
+}
+
+class GraphSiteClassificationUpdateCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.SITECLASSIFICATION_SET}`;
+  }
+
+  public get description(): string {
+    return 'Updates site classification configuration';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.usageGuidelinesUrl = typeof args.options.usageGuidelinesUrl !== 'undefined';
+    telemetryProps.guestUsageGuidelinesUrl = typeof args.options.guestUsageGuidelinesUrl !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
+      .then((): request.RequestPromise => {
+        const requestOptions: any = {
+          url: `${auth.service.resource}/beta/settings`,
+          headers: Utils.getRequestHeaders({
+            authorization: `Bearer ${auth.service.accessToken}`,
+            accept: 'application/json;odata.metadata=none'
+          }),
+          json: true
+        };
+
+        if (this.debug) {
+          cmd.log('Executing web request...');
+          cmd.log(requestOptions);
+          cmd.log('');
+        }
+
+        return request.get(requestOptions);
+      })
+      .then((res: { value: DirectorySetting[]; }): request.RequestPromise | Promise<void> => {
+        const unifiedGroupSetting: DirectorySetting[] = res.value.filter((directorySetting: DirectorySetting): boolean => {
+          return directorySetting.displayName === 'Group.Unified';
+        });
+
+        if (!unifiedGroupSetting ||
+          unifiedGroupSetting.length === 0) {
+          cb(new CommandError("There is no previous defined site classification which can updated."));
```

Comment:
This can be simplified to `return Promise.reject("There is no...")`. The rejected promise will be handled at the end including calling the callback with the error message.